src/pages/analytics/Analytics.js

```diff
@@ -147,14 +145,15 @@ class Analytics extends Page {
         return formData
     }
 
-    isAnalyzingTables = () => this.state.jobId && this.state.intervalId
+    isAnalyzingTables = () => !!this.state.intervalId
 
     startsPooling = () =>
         setInterval(() => {
             this.requestTaskSummary()
         }, PULL_INTERVAL)
 
     isJobInProgress = jobNotifications =>
+        // When the most recent job comes back as completed the whole process is done
```

Comment:
So should this not be `When all items in jobNotifications are completed the whole process is done`? Because this checks all jobNotifications.

Reply:
No that's not correct. This is the condition:
```javascript
isJobInProgress = jobNotifications =>
        jobNotifications.every(notification => !notification.completed)
```
So that means: job is in-progress as long as ALL items in the array have completed set to false.
And also it means: as soon as one item is marked completed `this.isJobInProgress()` will return `false`.

In practice, it seems that the item that is marked `completed: true` is always at index `0`. However, I am not sure if that's always the case, so I kept this `every` condition as it was...
So I guess the comment is not really mirroring the code 1:1 but the comment explains what happens, and the code is a way to deal with that behavior. Granted, the implementation might be a bit overly cautious but I would still just like to keep things this way.

Reply:
Ah yeah of course, sorry, you're right.